R/net.dat.R

```diff
@@ -0,0 +1,385 @@
+    out <- dat[["attr"]][[item]][indexes]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+add_attr <- function(dat, item) {
+  if (item %in% names(dat[["attr"]]))
+    stop(paste0("Cannot create the attribute '", item,
+               "': exists already"))
+
+  dat[["attr"]][[item]] <- rep(NA, length(dat$attr$active))
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+set_attr <- function(dat, item, value) {
+  if (!item %in% names(dat[["attr"]])) {
+    message(paste0("Creating attribute: '", item,
+               "' in the attributes list of the Master list object (dat)"))
+
+
+    dat <- add_attr(dat, item)
+  }
+
+  if (length(value) != length(dat$attr$active))
+    stop(paste0(
+      "When trying to edit the ", `item`, " nodale attribute: The size",
+       " of the `value` vector is not equal to the number of node in
+       the network. Expected: ", length(dat$attr$active), ", given : ",
+       length(value)))
+
+  dat[["attr"]][[item]] <- value
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_epi_list <- function(dat, item = NULL) {
+  if (is.null(item)) {
+    out <- dat[["epi"]]
+
+  } else {
+    missing_item <- setdiff(item,names(dat[["epi"]]))
+    if (length(missing_item) > 0)
+      stop(paste("There is no Epidemic output called",
+                 paste(missing_item, collapse = ", "),
+                 "in the Epidemic output list of the Master list object (dat)"))
+
+    out <- dat[["epi"]][item]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+get_epi <- function(dat, item, indexes = NULL) {
+  if (!item %in% names(dat[["epi"]]))
+      stop(paste("There is no Epidemic output called", item,
+                 "in the Epidemic output list of the Master list object (dat)"))
+
+  if (is.null(indexes)) {
+    out <- dat[["epi"]][[item]]
+
+  } else {
+    if (is.logical(indexes)) {
+      if (length(indexes) != dat$control$nsteps)
+        stop("(logical) `indexes` has to have a length equal to the number of
+              steps planned for for the simulation (control$nsteps)")
+    } else if(is.numeric(indexes)) {
+      if (any(indexes > dat$control$nsteps))
+        stop("Some (numeric) `indexes` are larger than the number of
+              steps planned for for the simulation (control$nsteps)")
+    } else {
+      stop("`indexes` must be logical, numeric, or NULL")
+    }
+
+    out <- dat[["epi"]][[item]][indexes]
+  }
+
+  return(out)
+}
+
+#' @rdname dat_get_set
+#' @export
+add_epi <- function(dat, item) {
+  if (item %in% names(dat[["epi"]]))
+    stop(paste("Cannot create the Epidemic outpout, ", item,
+               ": exists already"))
+
+  dat[["epi"]][[item]] <- rep(NA, dat$control$nsteps)
+
+  return(dat)
+}
+
+#' @rdname dat_get_set
+#' @export
+set_epi <- function(dat, item, value) {
+  if (!item %in% names(dat[["epi"]])) {
+    message(paste("Creating Epidemic output: '", item,
+               "'in the Epidemic output list of the Master list object (dat)"))
+
```

Comment:
I don't want these messages. They will just clutter the console if there are 50 of them. Please remove

Reply:
removed